clang/test/CIR/CodeGen/array-init.cpp

```diff
@@ -0,0 +1,38 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -std=c++17 -fclangir -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir %s
+
+typedef struct {
+  int a;
+  int b[2];
+} A;
+
+int bar() {
+  return 42;
+}
+
+void foo() {
+  A a = {bar(), {}};
+}
+// CHECK: %0 = cir.alloca !ty_A, !cir.ptr<!ty_A>, ["a", init]
+// CHECK: %1 = cir.alloca !cir.ptr<!s32i>, !cir.ptr<!cir.ptr<!s32i>>, ["arrayinit.temp", init]
```

Comment:
You need to use regexes for SSA values and struct definitions, see examples in `clang/test/CIR/CodeGen/abstract-cond.c` and eslewhere.

Reply:
updated